packages/crud/src/vaadin-crud.js

```diff
@@ -620,39 +616,21 @@ class Crud extends SlotMixin(ControllerMixin(ElementMixin(ThemableMixin(PolymerE
   }
 
   /** @protected */
-  get slots() {
-    // NOTE: order in which the toolbar buttons are listed matters.
-    return {
-      ...super.slots,
-      header: () => {
-        return document.createElement('h3');
-      },
-      form: () => {
-        return document.createElement('vaadin-crud-form');
-      },
-      'save-button': () => {
-        const button = document.createElement('vaadin-button');
-        button.id = 'save';
-        button.setAttribute('theme', 'primary');
-        return button;
-      },
-      'cancel-button': () => {
-        const button = document.createElement('vaadin-button');
-        button.id = 'cancel';
-        button.setAttribute('theme', 'tertiary');
-        return button;
-      },
-      'delete-button': () => {
-        const button = document.createElement('vaadin-button');
-        button.id = 'delete';
-        button.setAttribute('theme', 'tertiary error');
-        return button;
-      },
-    };
+  get _headerNode() {
+    return this._headerController && this._headerController.node;
   }
 
   constructor() {
     super();
+
+    this.__cancel = this.__cancel.bind(this);
```

Comment:
Moved these methods binding from `ready()` to the `constructor()` which is a more suitable place.
Also, to make the `ready()` method slightly shorter (it became quite long due to controllers).